README.md

```diff
@@ -1,27 +1,137 @@
+  exceptional,
+- the _monolith_ mode, provided as a plugin, generates a program using
+  [Monolith] that will test the module by comparing it against an
+  instrumented version of the module; see the [dedicated README] for
+  details.
+
+[Monolith]: https://gitlab.inria.fr/fpottier/monolith
+[dedicated README]: plugins/monolith/README.md
 
-**Disclamer:** This project is still experimental. No support will be provided
-at this point, and its behaviour is still unstable.
 
 ## Installation
 
+This project is still under development (standard warnings apply...),
+with no stable release yet.
+Consequently the easiest way to try it out is to `opam pin` it:
+
 ```
-opam pin add -y https://github.com/ocaml-gospel/gospel.git
 opam pin add -y https://github.com/ocaml-gospel/ortac.git
-opam install ortac
 ```
 
-## how to add a plugin
+It requires a yet-unreleased version of Gospel because Gospel evolved
+quite a lot since its 0.1.0 release.
+
+This repository contains the following OPAM packages:
+
+- `ortac.opam` which provides the `ortac` command-line tool and its
+  default mode,
+- `ortac-runtime.opam` which provides the support library for the code
+  generated by Ortac default mode,
+- `ortac-monolith.opam` which provides the Monolith plugin for the
+  `ortac` command-line tool,
+- `ortac-runtime-monolith.opam` which provides the support library for
+  the code generated with the Monolith plugin.
+
+You can install only some of those packages by explicitly mentionning
+which package you want to install, for instance:
+
+```
+$ opam pin add ortac https://github.com/ocaml-gospel/ortac.git
+```
+
+To check that it is correctly installed, simply run `ortac`. If you
+installed all packages, you should get:
+
+```
+$ ortac
+ortac: required COMMAND name is missing, must be either 'default' or 'monolith'.
+Usage: ortac COMMAND …
+Try 'ortac --help' for more information.
+```
+
+
+## Quick start
+
+or: How to use Ortac to test whether the specifications of a module
+hold. Using the Monolith plugin is probably the easiest option to
+start with.
+
+
+### Monolith plugin
+
+The Monolith plugin can generate a standalone executable that will try
+to falsify the Gospel specifications of a module by stress-testing the
+code. Look in the [dedicated README] for the Monolith plugin to see
+how it can be used.
+
+
+### Default mode
+
+The `default` mode can be used to generate a _wrapper_ module that
+will expose the same interface as the original module but
+instrumenting all function calls with assertions corresponding to the
+Gospel specifications.
+
+Let’s start with a module `lib.ml` containing some Gospel
+specifications, borrowed from Gospel documentation:
+
+```ocaml
+val euclidean_division: int -> int -> int * int
+(*@ q, r = euclidean_division x y
+    requires y > 0
+    ensures  x = q * y + r
+    ensures  0 <= r < y *)
+```
+
+Given that interface, `ortac default lib.mli` will display a code
+structured as such:
```

Comment:
The idea was to give a clue of what the generated code is doing, since it is fairly long but easy to generate and since code like:
```
Ortac_runtime.Gospelstdlib.(>) (Ortac_runtime.Gospelstdlib.integer_of_int y) (Ortac_runtime.Gospelstdlib.integer_of_int 0)
```
is quite a mouthful so it helps a lot if you know before reading it what this should be doing.

Reply:
I was thinking about adding the parenthesis, not the actual code.
I agree that it wouldn't be readable (nor helpful) to have the whole code.
I was just suggesting to be explicit and tell the reader that the comments are replaced by code doing what is said (but it may be a bit too much)